aten/src/ATen/cpu/vec/sve/vec_bfloat16.h

```diff
@@ -299,14 +299,18 @@ Vectorized<c10::BFloat16> inline operator/(
   return binary_operator_via_float(std::divides<Vectorized<float>>(), a, b);
 }
 
+inline Vectorized<BFloat16>::Vectorized() {
+  const short zero = 0;
+  values = svdup_n_bf16(*reinterpret_cast<const bfloat16_t*>(&zero));
```

Comment:
this has undefined behavior (because it's not on the list of permissible uses of reinterpret_cast). I think you could fix it with c10::bit_cast in this instance, but why not type `zero` as `const bfloat16_t` in the first place?

Reply:
`bfloat16_t` is a storage-only type, we can't convert any kind of 0 to it, though maybe if we use designated initialization it will be initialized with an all-zero bit pattern.

This does not have undefined behavior, and it occurs in many other files, which you can find with

```sh
rg "reinterpret_cast.*bfloat16_t ?\\*"
```

Reply:
Ok, I revised my opinion. This is UB, even though we do it dozens of times elsewhere, and I should use c10::bit_cast, which presumably uses memcpy